src/audio/smart_amp_test.c

```diff
@@ -440,6 +440,8 @@ static int smart_amp_copy(struct comp_dev *dev)
 		 avail_passthrough_frames);
 
 	buffer_lock(sad->feedback_buf, &feedback_flags);
+	dcache_invalidate_region(sad->feedback_buf->source,
```

Comment:
Better to extract to dedicated function along with the check for is_shared.

Reply:
good idea

Reply:
updated